Sources/NIOWebsocket/WebsocketOpcode.swift

```diff
@@ -0,0 +1,120 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//
+//  WebsocketOpcode.swift
+//  NIOWebsocket
+//
+//  Created by Cory Benfield on 16/02/2018.
+//
+
+/// An operation code for a websocket frame.
+public enum WebsocketOpcode {
+    case continuation
+    case text
+    case binary
+    case unknownNonControl(UInt8)
+    case connectionClose
+    case ping
+    case pong
+    case unknownControl(UInt8)
+
+    /// Create an opcode from the encoded representation.
+    ///
+    /// - parameters
+    ///     - encoded: The encoded representation of the opcode as an 8-bit integer.
+    ///          Must be no more than 4 bits large.
+    public init?(_ encoded: UInt8) {
+        guard encoded < 0x10 else {
+            return nil
+        }
+
+        switch encoded {
+        case 0x0:
+            self = .continuation
+        case 0x1:
+            self =  .text
+        case 0x2:
+            self = .binary
+        case 0x8:
+            self = .connectionClose
+        case 0x9:
+            self = .ping
+        case 0xA:
+            self = .pong
+        case let oc where (oc & 0x8) != 0:
+            self = .unknownControl(oc)
+        default:
+            self = .unknownNonControl(encoded)
+        }
+    }
+
+    /// Convert this opcode to the encoded 4-bit representation used on the wire.
+    ///
+    /// - returns: The encoded opcode as an 8-bit integer, with the opcode in the four
+    ///     least-significant bits. Returns `nil` if the opcode is not actually a valid
+    ///     websocket opcode.
+    public func toInt() -> UInt8? {
```

Comment:
should this be a property and not a method and be called `asInt` or something like that